src/views/List.jsx

```diff
@@ -1,15 +1,60 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
+import { useEffect } from 'react';
 
 export function List({ data }) {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [searchResults, setSearchResults] = useState([]);
+	// user type in search item - query
+	// as letters come in, filters items rendered on page
+	// set query to searchQuery
+	// render onto page search Results -> setting whatever is filtered into searchResults
+
+	function filterResults(query) {
+		return data.filter((item) => item.name.toLowerCase().includes(query));
+	}
+
+	useEffect(() => {
+		setSearchResults(filterResults(searchQuery));
+
+		// ignoring dependency array warning for now
+		// adding filterResults causes infinite re-render
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [searchQuery]);
+
+	function handleClick() {
+		setSearchResults([]);
+	}
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+			<form>
+				<label htmlFor="search-items">
+					Search Items:
+					<input
+						name="search-items"
+						id="search-items"
+						type="text"
+						placeholder="Search items"
+						value={searchQuery}
+						onChange={(e) => setSearchQuery(e.target.value)}
+					/>
+					{searchQuery ? (
```

Comment:
Nice addition of this 'clear search' button!

Reply:
I have only a naming suggestion - instead of handleClick it could be something reflecting its search clearing purpose. Or is it a convention that all button function names should start with 'handle'? Asking for a friend.

I like how the ternary operator is used in the button display

Reply:
handle is a convention but I agree it can be more descriptive since very easily there could be multiple handleClick like functions in this component --> like handleClearSearchResults

Reply:
Nice work on implementation and keeping the onChange really simple and straightforward!

Reply:
I've updated the handleClick to be `handleClearSearchQuery` since our other naming conventions were `searchQuery` and `setSearchQuery`. I can also go back and rename some of our other handleClicks that we have in other files because I agree, more verbose naming will be better as the app grows.